src/rfactor/rfactor.py

```diff
@@ -1,134 +1,281 @@
-from subprocess import check_call
-import os
+from functools import partial
+
+import numpy as np
+import pandas as pd
+
+from joblib import Parallel, delayed
 import multiprocessing as mp
 
-from pathlib import Path
-from dotenv import load_dotenv, find_dotenv
-from tqdm.contrib.concurrent import process_map
-from tqdm import tqdm
 
-import numpy as np
+TIME_BETWEEN_EVENTS = "6 hours"
+MIN_CUMUL_EVENT = 1.27
+
+class RFactorInputException(Exception):
+    """Raise when input data is not conform the rfactor required input format"""
+
+
+def rain_energy_per_unit_depth(rain):
+    """Calculate rain energy per unit depth according to Salles/Verstraeten
+
+    # TODO -> check the factor 6 -> linked to 10min interval? Is this requirement?
```

Comment:
see https://github.com/cn-ws/rfactor/issues/27, let's tackle this after the merge